openraft/src/core/admin.rs

```diff
@@ -73,15 +73,41 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
 }
 
 impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>> LeaderState<'a, D, R, N, S> {
+    #[tracing::instrument(level = "debug", skip(self))]
+    async fn add_learner_into_membership(&mut self, target: &NodeId) {
+        tracing::debug!(
+            "add_learner_into_membership target node {} into learner {:?}",
+            target,
+            self.nodes.keys()
+        );
+
+        let curr = &self.core.effective_membership.membership;
+        if curr.contains(target) {
+            tracing::debug!(
+                "target node {} is already a member or learner,cannot add as learner",
+                target
+            );
+            return;
+        }
```

Comment:
The result matters: If a learner already exists in the current config, it should notify the caller not to spawn a replication task for it.